model/src/main/kotlin/OrtResult.kt

```diff
@@ -189,9 +190,19 @@ data class OrtResult(
             { project -> project.id },
             { project ->
                 val pathExcludes = getExcludes().findPathExcludes(project, this)
+                val pathIncludes = getIncludes().findPathIncludes(project, this)
+
+                val isExcluded = if (getIncludes() == Includes.EMPTY) {
+                    // No includes are defined. It is excluded if it has path excludes.
+                    pathExcludes.isNotEmpty()
+                } else {
+                    // Some includes are defined. It is excluded if it has no path includes or has path excludes.
+                    pathIncludes.isEmpty() || pathExcludes.isNotEmpty()
+                }
```

Comment:
I don't like that the logic here gets more complicated because it is hard to test the code in this location. The logic should rather be inside of the `Includes` and `Excludes` classes and the condition here should be as simple as:

```kotlin
val definitionFilePath = getDefinitionFilePathRelativeToAnalyzerRoot(project)
val isExcluded = !getExcludes().isExcluded(definitionFilePath) && getIncludes().isIncluded(definitionFilePath)
```

And actually this code should go to a helper function that's easy to test.

But that's something that could be refactored separately, so I'm ok with the code for now.

Reply:
Yes, and as mentioned above, the function could be either a helper function or in `RepositoryConfiguration`. Therefore this complicated test would be done only in one location.

But I agree, let's refactor it later.